segmentation/utils/segmentation_utils.py

```diff
@@ -278,6 +278,86 @@ def combine_segmentation_masks(big_mask_xr, small_mask_xr, size_threshold,
         format="NETCDF3_64BIT")
 
 
+def find_nuclear_mask_id(nuc_segmentation_mask, cell_coords):
+    """Get the ID of the nuclear mask which has the greatest amount of overlap with a given cell
+
+    Args:
+        nuc_segmentation_mask: label mask of nuclear segmentations
+        cell_coords: list of coords specifying pixels that belong to a cell
+
+    Returns:
+        nuclear_mask_id: ID of the nuclear mask that overlaps most with cell. If not matches found,
+            returns None.
+    """
+
+    ids, counts = np.unique(nuc_segmentation_mask[tuple(cell_coords.T)], return_counts=True)
+
+    max_id = ids[np.argmax(counts)]
+
+    if max_id == 0:
+        if len(ids) == 1:
+            nuclear_mask_id = None
+        else:
+            non_zero_idx = ids > 0
+            ids, counts = ids[non_zero_idx], counts[non_zero_idx]
+
+            nuclear_mask_id = ids[np.argmax(counts)]
+    else:
+        nuclear_mask_id = max_id
+
+    return nuclear_mask_id
+
+
+def transform_expression_matrix(cell_data, transform, transform_kwargs={}):
+    """Transform an xarray of marker counts with supplied transformation
+
+    Args:
+        cell_data: xarray containing marker expression values
+        transform: the type of transform to apply. Must be one of ['size_norm', 'arcsinh']
+
+    Returns:
+        cell_data_norm: counts per marker normalized by cell size
+    """
+    valid_transforms = ['size_norm', 'arcsinh']
+
+    if transform not in valid_transforms:
+        raise ValueError('Invalid transform supplied')
+
+    # generate array to hold transformed data
+    cell_data_transformed = copy.deepcopy(cell_data)
+
+    # get start and end indices of channel data. We skip the 0th entry, which is cell size
+    channel_start = 1
+
+    # we include columns up to 'label', which is the first non-channel column
+    channel_end = np.where(cell_data.features == 'label')[0][0]
+
+    if transform == 'size_norm':
+
+        # get the size of each cell
+        cell_size = cell_data.values[:, :, 0:1]
+
+        # generate cell_size array that is broadcast to have the same shape as the channels
+        cell_size_large = np.repeat(cell_size, channel_end - channel_start, axis=2)
+
+        # Only calculate where cell_size > 0
+        cell_data_transformed.values[:, :, channel_start:channel_end] = \
+            np.divide(cell_data_transformed.values[:, :, channel_start:channel_end],
+                      cell_size_large, where=cell_size_large > 0)
+
+    elif transform == 'arcsinh':
+        linear_factor = transform_kwargs.get('linear_factor', 100)
+
+        # first linearly scale the data
+        cell_data_transformed.values[:, :, channel_start:channel_end] *= linear_factor
+
+        # arcsinh transformation
+        cell_data_transformed.values[:, :, channel_start:channel_end] = \
+            np.arcsinh(cell_data_transformed[:, :, channel_start:channel_end].values)
+
+    return cell_data_transformed
+
```

Comment:
Should this be split into two different functions?  I know the preprocessing for both functionalities is the same, but that could be put in an internal utility function.  That way, if more types of transformations are wanted later, we don't have to keep adding `transform` cases into this function

Another option in this line of thinking is having the `transform` argument be a function instead of a string, but that would probably be a little confusing to lab-users

Reply:
Or maybe we could pull out these transforms into their own function, and then just call them here, rather than defining the code here? Or do you think it's better to not have a high-level function at all, and instead pull our the redundant code into a low-level helper function that gets called at the beginning of each

Reply:
I like the idea of the high-level transform function, but if there are more transforms, we basically end up with a switch-case over what transform the user is asking for.  If we plan on having only these two transforms, then the first option you mentioned is probably best.  If we plan on having significantly more, then something along the lines of the second option.

Reply:
Got it. I'm not sure TBH, I think we'll have to see. In that case I'll leave it as is and we can decide which makes more sense as we continue to build it out